doc/source/whatsnew/v1.1.1.rst

```diff
@@ -46,6 +46,10 @@ Bug fixes
 - Bug in ``.groupby(..).rolling(..)`` where passing ``closed`` with column selection would raise a ``ValueError`` (:issue:`35549`)
 - Bug in :class:`DataFrame` constructor failing to raise ``ValueError`` in some cases when ``data`` and ``index`` have mismatched lengths (:issue:`33437`)
 
+**Reshaping**
+
+- Bug where :meth:`DataFrame.replace` and :meth:`Series.replace` would fail when replacement value was of a different dtype but equal to values in the original object (:issue:`34871`).
```

Comment:
can you move to 1.2

Reply:
this looks similar to regression reported in #35376 with open PR (with different fix)  #36444

Reply:
Yeah I think it's the same issue. If another fix avoids astyping to object I would think it's preferred.